### PR TITLE
Configre keybinds with keynames

### LINKED ISSUE #28: Replace hardcoded keycodes with key names
It would be more flexible to specify key name in config instead of key code.

### PR DESCRIPTION
**Pull request by TheHussky**
- Add key settings to config
- Use keysyms instead of keycodes
- Bind keysyms to keynames (hardcoded)

NOTE: Map of keynames:keysyms is hardcoded in `keys.hpp` due to limitations of xcb

### CODE REVIEW

File: src/config.hpp [hunk 1]
```diff
@@ -15,11 +16,13 @@ using color = const uint32_t;
...
 ///
 const uint dxp_width = 0;
-const uint dxp_height = 150;
+const uint dxp_height = 250;
 
 const int16_t dxp_x = 0; ///< X coordinate of window's top left corner
 const int16_t dxp_y = 0; ///< Y coordinate of window's top left corner
 
+const bool centering = false;///< Place dxp window centered on screen
+
```
Comment (mmskv) [inline src/config.hpp]: Whats this

File: src/config.hpp [hunk 2]
```diff
@@ -15,11 +16,13 @@ using color = const uint32_t;
 /// Actual window width will be equal to dxp_width + (dxp_padding * 2)
 ///
 const uint dxp_width = 0;
-const uint dxp_height = 150;
+const uint dxp_height = 250;
```
Comment (mmskv) [inline src/config.hpp]: Restore defaults pls

File: src/config.hpp [hunk 3]
```diff
@@ -50,4 +53,30 @@ const auto dxp_screenshot_period = std::chrono::seconds (1);
...
+/// View keynames.hpp for a list of supported keys
+/// and their names
+
+const std::vector<int> next_key = {Down, Right};
+
+const std::vector<int> prev_key =  {Left, Up};
+
+const std::vector<int> exit_key = {Escape};
+
+const std::vector<int> choose_key = {Return};
```
Comment (mmskv) [inline src/config.hpp]: Use autoformatter

File: src/keynames.hpp [hunk 4]
```diff
@@ -0,0 +1,1389 @@
+const int VoidSymbol=0xFFFFFF; //Does nothing
```
Comment (mmskv) [inline src/keynames.hpp]: Remove empty comments (or I can do this with couple of key presses in vim)

File: src/window.cpp [hunk 5]
```diff
@@ -330,15 +329,15 @@ window::handle_event (xcb_generic_event_t *event)
           {
             pres = pres == 0 ? desktops.size () - 1 : pres - 1;
           }
-        if (entr)
+        if (choose)
```
Comment (mmskv) [inline src/window.cpp]: better name would be `select`

File: src/xcb_util.hpp [hunk 6]
```diff
@@ -95,4 +95,8 @@ uint get_current_desktop (xcb_connection_t *c, xcb_window_t root);
 void ewmh_change_desktop (xcb_connection_t *c, xcb_window_t root,
                           uint destkop_id);
 
+xcb_keycode_t* get_keycodes(xcb_connection_t *c, xcb_keysym_t sym);
+
+bool check_key(xcb_connection_t *c, int received_key_code, std::vector<int> key_function);
```
Comment (mmskv) [inline src/xcb_util.hpp]: Document functions

File: src/xcb_util.cpp [hunk 7]
```diff
@@ -234,3 +234,42 @@ ewmh_change_desktop (xcb_connection_t *c, xcb_window_t root, uint destkop_id)
...
 }
+
+xcb_keycode_t *
+get_keycodes (xcb_connection_t *c, xcb_keysym_t sym) // const char * name)
+{
+  /**
+   * 1. Find a keysym for keyname in struct
+   * 2. Find possible keycodes for keysym
+   * 3. Return a vector of results.
+   */
```
Comment (mmskv) [inline src/xcb_util.cpp]: You should place docs above function definition

**Review by mmskv — CHANGES_REQUESTED** [review]
- Use autoformatter
- Reset config to defaults

File: src/config.hpp (hunk 1 above)
Comment (TheHussky) [reply src/config.hpp]: leftovers of my centering realization. will delete

File: src/keynames.hpp (hunk 4 above)
Comment (TheHussky) [reply src/keynames.hpp]: I think some of them are useful? Is it really needed?
TODO: add all different numbers that are not listed (arabic, farsi, etc.), add latin alphabet (this is for me)

File: src/window.cpp (hunk 5 above)
Comment (TheHussky) [reply src/window.cpp]: Oh, I agree. Will change.

File: src/xcb_util.hpp (hunk 6 above)
Comment (TheHussky) [reply src/xcb_util.hpp]: Will do

File: src/xcb_util.cpp (hunk 7 above)
Comment (TheHussky) [reply src/xcb_util.cpp]: Will correct

File: src/config.hpp (hunk 2 above)
Comment (TheHussky) [reply src/config.hpp]: Why tho? They'll be restored automatically when (if) we merge the branch, won't they?

File: src/config.hpp (hunk 3 above)
Comment (TheHussky) [reply src/config.hpp]: Done

File: src/xcb_util.cpp (hunk 7 above)
Comment (TheHussky) [reply src/xcb_util.cpp]: TODO: error handling
P.S. is it considered OK writing things like that in conversations?

File: src/keynames.hpp (hunk 4 above)
Comment (TheHussky) [reply src/keynames.hpp]: oh, EMPTY comments. Removed

File: src/config.hpp (hunk 2 above)
Comment (mmskv) [reply src/config.hpp]: no

File: src/xcb_util.cpp [hunk 8]
```diff
@@ -234,3 +234,36 @@ ewmh_change_desktop (xcb_connection_t *c, xcb_window_t root, uint destkop_id)
...
+  if (!key_symbols)
+    {
+      throw std::runtime_error("Couldn't allocate your keyboard symbols");
+    }
+  return xcb_key_symbols_get_keycode (key_symbols, sym);
+}
+
+bool
+check_key (xcb_connection_t *c, int received_key_code,
+           std::vector<int> key_function)
```
Comment (mmskv) [inline src/xcb_util.cpp]: Copypaste function docs here too

File: src/keynames.hpp [hunk 9]
```diff
@@ -0,0 +1,1509 @@
+const int VoidSymbol = 0xFFFFFF; // Does nothing
```
Comment (mmskv) [inline src/keynames.hpp]: mb remove all useless keys like VOID_SYMBOL etc?

File: src/config.hpp [hunk 10]
```diff
@@ -50,4 +53,30 @@ const auto dxp_screenshot_period = std::chrono::seconds (1);
...
+///
+/// Choose - select desktop
+/// (Enter by deafault)
+///
+/// View keynames.hpp for a list of supported keys
+/// and their names
+
+keys_for action_next = { Down, Right, latin_l, latin_k };
+
+keys_for action_prev = { Left, Up, latin_h, latin_j };
```
Comment (mmskv) [inline src/config.hpp]: Swap j and k buttons. j is next and k is prev

**Review by mmskv — CHANGES_REQUESTED** [review]
LGTM but too much junk in keynames.hpp

File: src/keynames.hpp (hunk 9 above)
Comment (TheHussky) [reply src/keynames.hpp]: I'd leave them just to be able to proudly say "Dxp recognizes ALL keys that modmap recognizes"

File: src/config.hpp (hunk 10 above)
Comment (TheHussky) [reply src/config.hpp]: will do

File: src/xcb_util.cpp (hunk 8 above)
Comment (TheHussky) [reply src/xcb_util.cpp]: ok

**Review by mmskv — APPROVED** [review]

**Pull request merged**